Odata-docs/webapi-8/fundamentals/overview.md

```diff
@@ -0,0 +1,582 @@
+
+Here are examples of things that can be defined in an EDM model:
+- Entity types
+- Complex types
+- Enums
+- Entity Sets
+- Singletons
+- Functions
+- Actions
+
+The EDM model also supports type inheritance and relationships between types.
+
+Entity sets, singletons, functions and actions can be exposed as endpoints that you can make requests to. Entity types, complex types and enum types define types of objects that can be used as inputs or outputs of those endpoints.
+
+To learn more about the OData data model, [visit this article](/odata/concepts/data-model).
+
+By default, an OData service exposes the schema on the `GET /$metadata` endpoint. Here's an example of what a response from this endpoint looks like:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
+    <edmx:DataServices>
+        <Schema Namespace="Lab01.Models" xmlns="http://docs.oasis-open.org/odata/ns/edm">
+            <EntityType Name="Order">
+                <Key>
+                    <PropertyRef Name="Id" />
+                </Key>
+                <Property Name="Id" Type="Edm.Int32" Nullable="false" />
+                <Property Name="Amount" Type="Edm.Decimal" Nullable="false" Scale="Variable" />
+            </EntityType>
+            <EntityType Name="Customer">
+                <Key>
+                    <PropertyRef Name="Id" />
+                </Key>
+                <Property Name="Id" Type="Edm.Int32" Nullable="false" />
+                <Property Name="Name" Type="Edm.String" />
+                <NavigationProperty Name="Orders" Type="Collection(Lab01.Models.Order)" />
+            </EntityType>
+        </Schema>
+        <Schema Namespace="Default" xmlns="http://docs.oasis-open.org/odata/ns/edm">
+            <EntityContainer Name="Container">
+                <EntitySet Name="Customers" EntityType="Lab01.Models.Customer" />
+            </EntityContainer>
+        </Schema>
+    </edmx:DataServices>
+</edmx:Edmx>
+```
+
+Clients can consume this metadata to know what types and endpoints the service exposes. Many OData clients use this metadata to generate strongly-typed client code. Tools like the [OData Connected Service](/odata/connectedservice/getting-started) and [OData CLI](/odata/odatacli/getting-started) can generate client code based on an OData schema.
+
+## Registering OData services
+
+It is possible to service multiple OData APIs in the same ASP.NET Core application.
+
+The first sample code in this document registers a single OData service using `AddRouteComponents(edmModel)`. This is hosted on the "default" route, i.e. on the root path `/`. This means that the endpoints on this OData service look like :
```

Comment:
```suggestion
The first sample code in this document registers a single OData service using `AddRouteComponents(IEdmModel)`. This is hosted on the "default" route, i.e. on the root path `/`. This means that the endpoints on this OData service look like :
```
The way to represent a method that accepts an argument is by including the argument type...